packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -669,16 +749,60 @@ class HeapTreeViewState extends State<HeapTree>
           ),
         ),
         const Spacer(),
-        Text(
-          controller.monitorTimestamp == null
-              ? 'No allocations tracked'
-              : 'Allocations Tracked at ${MemoryController.formattedTimestamp(controller.monitorTimestamp)}',
-          style: themeData.colorScheme.italicTextStyle,
+        updateCircle,
+      ],
+    );
+  }
+
+  Widget textWidgetWithUpdateCircle(
+    String text, {
+    TextStyle style,
+    double size,
+  }) {
+    final textWidth = textWidgetWidth(text, style: style);
+
+    return Stack(
+      children: [
+        Positioned(
+          child: Container(
+            child: Text(text, style: style),
+            width: textWidth + 10,
+          ),
+        ),
+        Positioned(
+          right: 0,
+          child: Container(
+            alignment: Alignment.topRight,
+            width: size,
+            height: size,
+            child: const Icon(Icons.circle, size: 0),
+            decoration: BoxDecoration(
+              shape: BoxShape.circle,
+              color: Colors.blue[400],
+            ),
+          ),
         ),
       ],
     );
   }
 
+  double textWidgetWidth(String message, {TextStyle style}) {
+    // self-defined constraint
+    const constraints = BoxConstraints(
+      maxWidth: 800.0, // maxwidth calculated
```

Comment:
comment seems incomplete. where is this calculated from?

Reply:
The message is not > 800 pixels in this usage.  However, I did add a TODO and a check to log a warning if this occurs.